src/NServiceBus.Core/Unicast/Messages/MessageMetadataRegistry.cs

```diff
@@ -165,15 +164,18 @@ static IEnumerable<Type> GetHandledMessageTypes(Type messageHandlerType)
 
     MessageMetadata RegisterMessageType(Type messageType)
     {
+        if (messageType.IsGenericType)
+        {
+            // This is not an error because in most cases it will work, but it's still not supported should issues arise
+            Logger.Warn($"Generic messages types are not supported. Consider converting {messageType.AssemblyQualifiedName} to a dedicated, simple type");
```

Comment:
How about we do log debug only?

Reply:
Not super necessary but would it be possible with little effort to write a unit test?

Reply:
We have decided not to write a test but switch the level to debug only to hint to developers that there are better ways to design their messages.